compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1283,6 +1301,15 @@ private boolean isArrayTypeAssignableToTupleType(BArrayType source, BTupleType t
                 return false;
             }
 
+            if (target.tupleTypes.stream().allMatch(this::isNoType)) {
```

Comment:
Seems we are iterating two times here.. first we iterate at line number 1264, then here again, what if we refactor isNoType() to deal with BArrayType also and return isNoType and done with a single iteration?

Reply:
Fixed in 10dc20b